.github/workflows/main.yaml

```diff
@@ -0,0 +1,91 @@
+name: CI
+
+on:
+  push:
+
+jobs:
+  lint-report:
+    name: Lint report
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install tox
+        run: pip install tox
+      - name: Run tests using tox
+        run: tox -e lint
+
+  static-analysis:
+    name: Static analysis
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install tox
+        run: pip install tox
+      - name: Run tests using tox
+        run: tox -e static
+
+  unit-tests-with-coverage:
+    name: Unit tests
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install tox
+        run: pip install tox
+      - name: Run tests using tox
+        run: tox -e unit
+
+  integration-tests:
+    name: Integration tests
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install tox
+        run: pip install tox
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: microk8s
+      - name: Run tests using tox
+        run: tox -e integration
+      - name: Archive Tested Charm
+        uses: actions/upload-artifact@v3
+        if: ${{ github.event_name == 'push' }}
+        with:
+          name: tested-charm
+          path: .tox/**/route53-lego-operator_ubuntu-20.04-amd64.charm
+          retention-days: 5
+  
+  publish-charm:
```

Comment:
Charm publishing shouldn't happen at every push.

Reply:
And make sure the charm is registered and all before merging to main to make sure it gets automatically published.

Reply:
I followed the same pattern we have in other charms, for example [tls-operator](https://github.com/canonical/tls-certificates-operator/blob/main/.github/workflows/main.yml).
Or am I missing something? :)